pkg/build/builder/daemonless.go

```diff
@@ -57,8 +130,18 @@ func pullDaemonlessImage(sc types.SystemContext, store storage.Store, imageName
 		return fmt.Errorf("error parsing image name to pull %s: %v", "docker://"+imageName, err)
 	}
 
+	credsFile, err := mergeNodeCredentials(dockercfg.GetDockerConfigPath(searchPaths))
+	if err != nil {
+		return err
+	}
+	defer func() {
+		if err := os.Remove(credsFile); err != nil {
```

Comment:
Would it be better to create the tempfile in this function, and pass it to mergeNodeCredentials, so that you can make sure that it gets deleted with a defer?

Reply:
Not sure if I followed you, if that is the case let me know. I have implemented it, PTAL.